server/base/src/main/java/org/apache/accumulo/server/log/WalStateManager.java

```diff
@@ -98,10 +102,16 @@ private String root() {
   // Tablet server exists
   public void initWalMarker(TServerInstance tsi) throws WalMarkerException {
     byte[] data = new byte[0];
-    try {
-      zoo.putPersistentData(root() + "/" + tsi.toString(), data, NodeExistsPolicy.FAIL);
-    } catch (KeeperException | InterruptedException e) {
-      throw new WalMarkerException(e);
+    while (true) {
+      try {
+        zoo.putPersistentData(root() + "/" + tsi.toString(), data, NodeExistsPolicy.FAIL);
+        break;
+      } catch (NoNodeException e) {
+        log.info("WAL parent node does not exist (upgrade may be in progress) : " + e.getMessage());
```

Comment:
@ShawnWalker   made a very nice suggestion irc.  Create the WALs node instead of waiting.  This breaks this invisible dependency on code in the master.